test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -60,5 +60,97 @@ public void FirstExperienceWithML()
                 PetalLength = 5.1f
             });
         }
+
+
+        [Fact]
+        public void SimpleIrisDescisionTrees()
+        {
+            var env = new ConsoleEnvironment();
+            string dataPath = "iris.txt";
+            // Create reader with specific schema.
+            var dataReader = TextLoader.CreateReader(env, dataPath,
+                ctx => (
+                    label: ctx.LoadText(0),
+                    sepalWidth: ctx.LoadFloat(1),
+                    sepalLength: ctx.LoadFloat(2),
+                    petalWidth: ctx.LoadFloat(3),
+                    petalLength: ctx.LoadFloat(4)
+                ));
+
+            var pipeline = dataReader.MakeEstimator()
+                .Append(row => (
+                    // Convert string label to key.
+                    label: row.label.Dictionarize(),
+                    // Concatenate all features into a vector.
+                    features: row.sepalWidth.ConcatWith(row.sepalLength, row.petalWidth, row.petalLength)))
+                .Append(row => (
+                    label: row.label,
+                    prediction: row.label.PredictFastTreeMultiClass(row.features)));
+
+            // Load the data into the system.
+            var data = dataReader.Read(dataPath);
+            var model = pipeline.Fit(data);
+
+            var predictions = model.Transform(dataReader.Read(testDataPath));
+            var metrics = MultiClassEvaluator.Evaluate(predictions, row => row.label, row => row.prediction);
+        }
+
+        [Fact]
+        public void TwitterSentimentAnalysis()
+        {
+            var env = new ConsoleEnvironment();
```

Comment:
I know we generally need `env`, but does it add any value in simple scenarios where it doesn't do anything?

Reply:
It is assuredly doing something. You're using it to create the text loader, and from there all subsequent components spawned from that object, which is to say, all of them.

Reply:
@TomFinley - The ConsoleEnvironment looks not clear to me. If we need the TlcEnvironment (ConsoleEnvironment) in most cases, in order to create a DataReader:

var dataReader = TextLoader.CreateReader(env, dataPath,...

Does it make sense to call it ConsoleEnvironment if I run the build/training code from a different type of application like an ASP.NET Core Web app (or a WinForms, WPF if using .NET Framework)?

Why don't we simply call it "Environment" or "MLNetEnvironment"?

Or is the TlcEnvironment (ConsoleEnvironment) only used in Console apps and if running a build/train model code in a non-console app I don't need the "env" object in order to create a DataReader?